public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -0,0 +1,585 @@
+    }
+
+    setOpenedMenuGroup(value);
+    setShowSubMenu(true);
+    const group = optionGroups.find((g) => {
+      return g.value === value;
+    });
+    if (group) {
+      setSubMenuOptions(group.options);
+    }
+  };
+
+  const onSelectedBuiltinRoleChange = (newRole: OrgRole) => {
+    setSelectedBuiltInRole(newRole);
+
+    let roles = builtInRoles[newRole];
+    if (newRole === OrgRole.Admin) {
+      roles = roles.concat(builtInRoles[OrgRole.Editor]);
+      roles = roles.concat(builtInRoles[OrgRole.Viewer]);
+    } else if (newRole === OrgRole.Editor) {
+      roles = roles.concat(builtInRoles[OrgRole.Viewer]);
+    }
+    setSelectedBuiltInRoles(roles);
+  };
+
+  const onClearInternal = async () => {
+    if (onClear) {
+      onClear();
+    }
+    setSelectedOptions([]);
+  };
+
+  const onClearSubMenu = () => {
+    const options = selectedOptions.filter((role) => {
+      const groupName = getRoleGroup(role);
+      return groupName !== openedMenuGroup;
+    });
+    setSelectedOptions(options);
+  };
+
+  const onUpdateInternal = () => {
+    const selectedCustomRoles: string[] = [];
+    for (const key in selectedOptions) {
+      const roleUID = selectedOptions[key]?.uid;
+      selectedCustomRoles.push(roleUID);
+    }
+    onUpdate(selectedBuiltInRole, selectedCustomRoles);
+  };
+
+  return (
+    <div className={cx(styles.menu, customStyles.menuWrapper)}>
+      <div className={customStyles.menu} aria-label="Role picker menu">
+        <CustomScrollbar autoHide={false} autoHeightMax="300px" hideHorizontalTrack>
+          <div className={customStyles.menuSection}>
+            <div className={customStyles.groupHeader}>Built-in roles</div>
+            <RadioButtonGroup
+              className={customStyles.builtInRoleSelector}
+              options={BuiltinRoleOption}
+              value={selectedBuiltInRole}
+              onChange={onSelectedBuiltinRoleChange}
+              fullWidth={true}
+            />
+          </div>
+          {showGroups && !!optionGroups.length ? (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {optionGroups.map((option, i) => (
+                  <RoleMenuGroupOption
+                    data={option}
+                    key={i}
+                    isSelected={groupSelected(option.value) || groupPartiallySelected(option.value)}
+                    partiallySelected={groupPartiallySelected(option.value)}
+                    onChange={onGroupChange}
+                    onClick={onMenuGroupClick}
+                  />
+                ))}
+              </div>
+            </div>
+          ) : (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {fixedRoles.map((option, i) => (
```

Comment:
There should also be a check for `fixedRoles` length, otherwise the header is still shown when there's no fixed roles available when doing the search.